include/openPMD/IO/JSON/JSONIOHandlerImpl.hpp

```diff
@@ -441,12 +441,7 @@ namespace openPMD
                 const Parameter< Operation::WRITE_DATASET > & parameters
             );
 
-            template< int n >
-            void operator()(
-                nlohmann::json & json,
-                const Parameter< Operation::WRITE_DATASET > & parameters
-            );
-
+            std::string errorMsg = "JSON: writeDataset";
```

Comment:
Cool, looks nice to use :)

You can `const` those members, I guess.
```suggestion
            std::string const errorMsg = "JSON: writeDataset";
```

Reply:
Hm, to be fair I'm not a huge fan of making non-static members `const` since that automatically deletes any `operator=` members. Since there are places where I store instances of these structs inside other structs (the ADIOS2 backend for example), I would prefer not to have some inconspicuous member silently delete those operators..

Reply:
Oh I see, agreed.